src/componets/calculator.js

```diff
@@ -1,29 +1,37 @@
-import React from 'react';
+import React, { useRef } from 'react';
+import calculate from '../logic/calculate';
 
 function Calculator() {
+  const display = useRef(null);
+  let result = { total: null, next: null, operation: null };
+  const getButtonName = (e) => {
+    const buttonName = e.target.textContent;
+    result = calculate(result, buttonName);
+    display.current.textContent = result.next || result.operation || result.total || '0';
+  };
```

Comment:
- [ ] Kindly use `useState` to handle state here and not `useRef` as you did. `useState` is designed specifically for managing state in React components while `useRef` is typically used for accessing and manipulating DOM elements.

Reply:
fixed